src/nedap/speced/def/impl/spec_assertion.clj

```diff
@@ -0,0 +1,34 @@
+(ns nedap.speced.def.impl.spec-assertion
+  (:require
+   [clojure.test :as test]
+   [nedap.speced.def.impl.parsing :refer [extract-specs-from-metadata]]
+   [nedap.utils.spec.api :refer [check!]])
+  (:import
+   (clojure.lang ExceptionInfo)))
+
+(defn spec-assertion-thrown? [clj? msg spec-sym body]
+  {:pre [(check! boolean?    clj?
+                 some?       spec-sym
+                 sequential? body)]}
+  (let [reporter (if clj?
+                   `test/do-report
+                   'cljs.test/do-report)]
+    `(try
+       (with-out-str ;; silently execute body
+         ~@body)
+       (~reporter {:type :fail, :message ~msg :expected '~spec-sym, :actual "no spec failure"})
+       (catch ~(if clj?
+                 `ExceptionInfo
+                 'ExceptionInfo) e#
+         (let [spec#          (:spec (ex-data e#))
```

Comment:
```suggestion
         (let [spec#           (:spec (ex-data e#))
```

Reply:
The almost-alignment was accidental, in reality we rarely align `let` I think (vs. hashmaps)?

Will unalign